src/lib/support/tests/TestTlvToJson.cpp

```diff
@@ -0,0 +1,273 @@
+                      "{\n"
+                      "   \"value\" : null\n"
+                      "}\n");
+
+    Clusters::TestCluster::Structs::SimpleStruct::Type structVal;
+    structVal.a = 20;
+    structVal.b = true;
+    structVal.d = byteBuf;
+    structVal.e = charSpan;
+    structVal.g = 1.0;
+    structVal.h = 1.0;
+
+    EncodeAndValidate(structVal,
+                      "{\n"
+                      "   \"value\" : {\n"
+                      "      \"0\" : 20,\n"
+                      "      \"1\" : true,\n"
+                      "      \"2\" : 0,\n"
+                      "      \"3\" : \"AQIDBP/+mYjdzQ==\",\n"
+                      "      \"4\" : \"hello\",\n"
+                      "      \"5\" : 0,\n"
+                      "      \"6\" : 1.0,\n"
+                      "      \"7\" : 1.0\n"
+                      "   }\n"
+                      "}\n");
+
+    uint8_t int8uListData[] = { 1, 2, 3, 4 };
+    DataModel::List<uint8_t> int8uList;
+
+    int8uList = int8uListData;
+
+    EncodeAndValidate(int8uList,
+                      "{\n"
+                      "   \"value\" : [ 1, 2, 3, 4 ]\n"
+                      "}\n");
+
+    Clusters::TestCluster::Structs::SimpleStruct::Type structListData[2] = { structVal, structVal };
+    DataModel::List<Clusters::TestCluster::Structs::SimpleStruct::Type> structList;
+
+    structList = structListData;
+
+    EncodeAndValidate(structList,
+                      "{\n"
+                      "   \"value\" : [\n"
+                      "      {\n"
+                      "         \"0\" : 20,\n"
+                      "         \"1\" : true,\n"
+                      "         \"2\" : 0,\n"
+                      "         \"3\" : \"AQIDBP/+mYjdzQ==\",\n"
+                      "         \"4\" : \"hello\",\n"
+                      "         \"5\" : 0,\n"
+                      "         \"6\" : 1.0,\n"
+                      "         \"7\" : 1.0\n"
+                      "      },\n"
+                      "      {\n"
+                      "         \"0\" : 20,\n"
+                      "         \"1\" : true,\n"
+                      "         \"2\" : 0,\n"
+                      "         \"3\" : \"AQIDBP/+mYjdzQ==\",\n"
+                      "         \"4\" : \"hello\",\n"
+                      "         \"5\" : 0,\n"
+                      "         \"6\" : 1.0,\n"
+                      "         \"7\" : 1.0\n"
+                      "      }\n"
+                      "   ]\n"
+                      "}\n");
+}
+
+int Initialize(void * apSuite)
+{
+    VerifyOrReturnError(chip::Platform::MemoryInit() == CHIP_NO_ERROR, FAILURE);
+    return SUCCESS;
+}
+
+int Finalize(void * aContext)
+{
+    (void) gStore.Release();
+    chip::Platform::MemoryShutdown();
+    return SUCCESS;
+}
+
+const nlTest sTests[] = { NL_TEST_DEF("TestConverter", TestConverter), NL_TEST_SENTINEL() };
+
+} // namespace
+
+nlTestSuite theSuite = { "TlvJson", sTests, Initialize, Finalize };
```

Comment:
This should be in the anonymous namespace, too. Otherwise, it conflicts with other tests when all tests are linked into a single binary.